devito/types/array.py

```diff
@@ -215,6 +215,19 @@ def _make_pointer(self, dim):
         return PointerArray(name='p%s' % self.name, dimensions=dim, array=self)
 
 
+class ArrayMapped(Array):
+
+    _C_structname = 'array'
+    _C_field_data = 'data'
+    _C_field_nbytes = 'nbytes'
+    _C_field_dmap = 'dmap'
+
+    _C_ctype = POINTER(type(_C_structname, (Structure,),
```

Comment:
you see? a ney symbolic type, `ArrayMapped`, but the only thing that needs to be defined is the `_C_ctype`. This is enough to create a `ctype`, a `str` representing its C type, etc etc.